File: library/src/main/java/com/nshmura/strictmodenotifier/StringModeConfig.java
package com.nshmura.strictmodenotifier;

import android.content.Context;
import android.content.SharedPreferences;

class StringModeConfig {
  private static final String NAME = "strictmode";
  private static final String KEY = "config";

  private final Context context;

  public static StringModeConfig from(Context context) {
    return new StringModeConfig(context);
  }

  StringModeConfig(Context context) {
    this.context = context;
  }

  public boolean isEnabled() {
    return getPrefs().getBoolean(KEY, true);
  }

  public void enable(boolean enabled) {
    getPrefs().edit().putBoolean(KEY, enabled).apply();
  }

  public void toggle() {
    enable(!isEnabled());
  }

  private SharedPreferences getPrefs() {
    return context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
  }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/StringModeConfig.kt
package io.github.juanimoli.strictmode.notifier

import android.content.Context
import android.content.SharedPreferences

internal class StringModeConfig(private val context: Context) {
    val isEnabled: Boolean
        get() = prefs.getBoolean(KEY, true)

    fun enable(enabled: Boolean) {
        prefs.edit().putBoolean(KEY, enabled).apply()
    }

    fun toggle() {
        enable(!isEnabled)
    }

    private val prefs: SharedPreferences
        get() = context.getSharedPreferences(NAME, Context.MODE_PRIVATE)

    companion object {
        private const val NAME = "strictmode"
        private const val KEY = "config"

        fun from(context: Context): StringModeConfig {
            return StringModeConfig(context)
        }
    }
}
